### PR TITLE
First edition of identifier test

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: mat_dp_pipeline/data_sources/country_sets/country_set.py [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
 import pandas as pd
 
+from .identifier import Identifier
+
 __all__ = ["CountrySet"]
 
 
+def get_none():
+    return None
+
```
Comment (mark-todd) [inline mat_dp_pipeline/data_sources/country_sets/country_set.py:12]: Remove this

File: mat_dp_pipeline/data_sources/country_sets/__init__.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
-from .country_set import CountrySet
+from .country_set import CountrySet, CustomCountry
 from .create_country_to_path import CountrySets
 from .source_with_countries import SourceWithCountries
 
-__all__ = ["CountrySet", "CountrySets", "SourceWithCountries"]
+__all__ = ["CountrySet", "CountrySets", "SourceWithCountries", "CustomCountry"]
```
[the remark this answers is not in the document]
Comment (mark-todd) [reply mat_dp_pipeline/data_sources/country_sets/__init__.py:5]: Above imports?

File: mat_dp_pipeline/data_sources/country_sets/create_country_to_path.py [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
             if isinstance(v, CountrySet):
-                all_countries = v.all_countries
-                for country in all_countries:
+                for country, country_name in v.all_countries(
+                    country_code_df, identifier
+                ):
                     if country in final_dict:
                         raise ValueError(
-                            f"Overlapped disjoint set detected for {country}"
+                            f"Overlapped disjoint set detected for {country} location is already {final_dict[country]}"
```
Comment (mkcinu) [inline mat_dp_pipeline/data_sources/country_sets/create_country_to_path.py:26]: Haha that's a funny. COmment. Not an issue, but Overlapped disjoint sounds funny ;]

File: mat_dp_pipeline/data_sources/mat_dp_db.py [hunk 4]
```diff
@@ -3,10 +3,77 @@
...
+    "General": "/",
+    "North America": "/North America",
+    "Central and South America": "/Central and South America",
+    "CHL": "/Central and South America/Chile",
+    "Africa": "/Africa",
+    "Europe": "/Europe",
+    "Middle East and Central Asia": "/Middle East and Central Asia",
+    "South and East Asia": "/South and East Asia",
+    "Oceania": "/Oceania",
+    "NM": "/Africa/Namibia",
```
Comment (mkcinu) [inline mat_dp_pipeline/data_sources/mat_dp_db.py:23]: Would be good to add a comment that NM isn;t a proper alpha-2, so we have to add it here.

File: mat_dp_pipeline/data_sources/tmba.py [hunk 5]
```diff
@@ -69,11 +75,13 @@ def __call__(self, output_dir: Path) -> None:
         # TODO: move this. to __init__?
         grouping = ["country", "parameter"]
         for key, targets_frame in targets.groupby(grouping):
-            # TODO: make some assertion that key[0] is a country. In __init__?
-            path = (location_to_path(key[0]),) + key[1:]
+            if key[0] == "NM":
```
Comment (mkcinu) [inline mat_dp_pipeline/data_sources/tmba.py:78]: Also good to add a comment, it isn't obvious.

File: mat_dp_pipeline/sdf/create_sdf.py [hunk 6]
```diff
@@ -69,13 +69,15 @@ def create_sdf(
...
-                    metadata.main_label = (
-                        main_label
-                        if isinstance(main_label, str)
-                        else main_label.main_label
-                    )
-                elif has_single_target_source:
-                    metadata.main_label = targets.main_label
+                    if isinstance(main_label, str):
+                        metadata.main_label = main_label
+                    elif main_label is not None and main_label.main_label is not None:
```
Comment (mkcinu) [inline mat_dp_pipeline/sdf/create_sdf.py:74]: `main_label is not None` is True for sure given line 71

File: mat_dp_pipeline/data_sources/mat_dp_db.py (hunk 4 above)
Comment (mark-todd) [reply mat_dp_pipeline/data_sources/mat_dp_db.py:23]: woops this one was actually a mistake

File: mat_dp_pipeline/data_sources/tmba.py (hunk 5 above)
Comment (mark-todd) [reply mat_dp_pipeline/data_sources/tmba.py:78]: agreed

File: mat_dp_pipeline/sdf/create_sdf.py [hunk 7]
```diff
@@ -69,13 +69,15 @@ def create_sdf(
...
+                        metadata.main_label = main_label.main_label
+                else:
+                    if intensities.main_label is not None:
```
[the remark this answers is not in the document]
Comment (mark-todd) [reply mat_dp_pipeline/sdf/create_sdf.py:77]: priority to intensities over indicators? Seemed like a natural priority to me - one of them has to have it so...

File: mat_dp_pipeline/sdf/create_sdf.py (hunk 6 above)
Comment (mark-todd) [reply mat_dp_pipeline/sdf/create_sdf.py:74]: yep

**Pull request merged**